src/main/notifications/index.ts

```diff
@@ -27,6 +27,11 @@ class NotificationManager {
     private upgradeNotification?: NewVersionNotification;
     private restartToUpgradeNotification?: UpgradeNotification;
 
+    constructor() {
+        // Run this very early to perform an early permission check
+        getDoNotDisturb();
```

Comment:
Will this trigger a permission modal before getting a notification?

Reply:
Yes, it shows the macOS one that people were missing.